### PR TITLE
Added Documentation for Windows WSL2 Client Setup

### LINKED ISSUE #5119: [Feature] Support WSL2!
## Summary

Appsmith currently doesn't support WSL2. This is prohibitive for the growing WSL community to use it and build on it/for it!

## Motivation

My own self not being able to install Appsmith, since I'm on WSL.

## Additional Context

WSL is Windows Subsystem for Linux, a newly launched capability to run Linux distributions natively inside Windows. WSL2 is the latest version of this feature and is now GA for all Windows 10 users. There is a growing community around WSL2.

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description

Documentation for helping setup Client environment in Windows WSL2

Fixes #5119 

## Type of change

Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by rahulramesha** [comment]
@somangshu please let me know your thoughts

File: contributions/ClientSetup.md [hunk 1]
```diff
@@ -107,6 +107,24 @@ node versions to be used in different projects. Check below for installation and
...
+In the above Docker Desktop setup instructions, make sure to:
+1. Set WSL Distro to run in WSL2 mode.
+2. Enable integration with the WSL Distro in Docker Desktop.
+3. Install [Remote-WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) extension in VSCode.
+
+And finally, you can open the folder in VSCode with WSL by,
+1. Clicking on Green button on the Bottom Left corner in VSCode
+2. Selecting **Reopen Folder in WSL**
+
+After this You can continue Setting up from [here](#pre-requisites). But make sure to have `Nodejs` and `npm` installed on the WSL Distro.
```
Comment (somangshu) [inline contributions/ClientSetup.md:126]: When we say `nodejs` and `npm` is required, Can we point them to possible documentation that shows how to check this and then install if missing

**Review by somangshu — DISMISSED** [review]
LGTM

**Comment by jsartisan** [comment]
@somangshu @rahulramesha One thing to add is: Make sure that project files are inside the ubuntu filesystem instead of the windows filesystem.

File: contributions/ClientSetup.md (hunk 1 above)
Comment (rahulramesha) [reply contributions/ClientSetup.md]: Moved that up and added a Documentation link.

**Comment by rahulramesha** [comment]
@jsartisan added that as well, please check once more.

**Comment by jsartisan** [comment]
LGTM!

**Review by jsartisan — APPROVED** [review]

**Review by somangshu — APPROVED** [review]
LGTM

**Pull request merged**